### PR TITLE
DGS-24644: Fix scan hang on offset gaps; protect unscannable topics from deletion

### PR DESCRIPTION
**Pull request by GunalKupta**
**Jira:** [DGS-24644](https://confluentinc.atlassian.net/browse/DGS-24644)

## What

Fixes a customer-reported scan hang and hardens `scan`/`delete` against incomplete scans.

- **Scan no longer hangs on offset gaps.** Terminate on partition EOF (`enable.partition.eof=true`, `isolation.level=read_committed`) instead of comparing the last offset to the high watermark — a transaction control record or compaction can leave a gap the old comparison never crossed, hanging forever.
- **An unscannable topic no longer aborts the whole run, and can't cause a bad delete.** Scanned topics still produce a partial manifest; subjects tied to a failed topic are marked `blocked_unverified` and excluded from deletion.
- **`--scan-timeout` (default 5m)** makes the per-topic deadline configurable for large topics.
- **Fail fast on auth/authz errors** (topic/cluster/group authz, SASL, authentication) instead of polling until the deadline.
- **`delete` refuses a manifest with unverified topics** unless `--allow-unverified` is passed (deliberately *not* bypassed by `--force`).

## Key safety property

Any scan deadline/error returns an *error*, so "scanned successfully" always means "fully scanned" — an incomplete scan can never make an in-use schema look unused and get deleted.

`go build`, `go vet`, `go test ./...` all pass.

[DGS-24644]: https://confluentinc.atlassian.net/browse/DGS-24644?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/safety.go [hunk 1]
```diff
@@ -67,6 +67,24 @@ func AnalyzeCandidates(schemas []SchemaInfo, activeSchemas map[int32]int, platfo
...
+	if len(unverifiedSubjects) == 0 {
+		return candidates
+	}
+	for i := range candidates {
+		if unverifiedSubjects[candidates[i].Subject] {
+			candidates[i].Status = StatusBlockedUnverified
+			candidates[i].BlockReasons = append(candidates[i].BlockReasons,
+				"Topic could not be scanned; schema usage could not be verified")
+		}
+	}
```
[the remark this answers is not in the document]
Comment (GunalKupta) [reply pkg/safety.go:88]: Good catch — fixed in 0aead77. `MarkUnverified` now keeps an already-blocked candidate's more specific status (e.g. `blocked_by_references`) and only sets `blocked_unverified` when the candidate isn't already blocked, appending the unverified reason in both cases. A *warned* candidate is still upgraded to blocked, since an unverifiable topic must never be deletable. Added tests for the already-blocked and warned cases.

File: pkg/utils.go [hunk 2]
```diff
@@ -43,12 +43,17 @@ func VerifySubject(subject string) bool {
 	return IsValueSchema(subject) || IsKeySchema(subject)
 }
 
+// TopicFromSubject strips the context prefix and the -value/-key suffix to
+// recover the topic name for a TopicNameStrategy subject.
+func TopicFromSubject(subject string) string {
+	return strings.TrimSuffix(strings.TrimSuffix(GetRawSubject(subject), "-value"), "-key")
+}
```
[the remark this answers is not in the document]
Comment (GunalKupta) [reply pkg/utils.go:55]: Agreed — fixed in 0aead77. `TopicFromSubject` now strips exactly one suffix (`-value` if present, otherwise `-key`), so a topic literally named `orders-key` (subject `orders-key-value`) resolves back to `orders-key` rather than `orders`. Added a test covering the `-key`/`-value` topic-name edge cases.

File: cmd/root.go [hunk 3]
```diff
@@ -322,46 +348,57 @@ func scanTopicsForActiveSchemas(topics []pkg.TopicWithClusterInfo, platform pkg.
...
+
+// mergeActiveSchemas OR-merges per-topic schema results and returns the names of
+// topics that failed to scan.
+func mergeActiveSchemas(results []scanResult) (map[int32]int, []string) {
+	activeSchemas := make(map[int32]int)
+	var failedTopics []string
+	for _, r := range results {
+		if r.err != nil {
+			failedTopics = append(failedTopics, r.topic)
 			continue
```
[the remark this answers is not in the document]
Comment (GunalKupta) [reply cmd/root.go:425]: Fixed in 0aead77. `mergeActiveSchemas` now de-duplicates `failedTopics` preserving first-seen order, so a topic that fails across multiple clusters appears once in the output and the manifest's `unverified_topics`. Added a cross-cluster duplicate test.

**Review by lei-tianyu — APPROVED** [review]
